### PR TITLE
feat(manager/bazel): recognize Bazel rules that begin with an underscore

### PR DESCRIPTION
**Pull request by zakcutner**
## Context and changes

When using Bazel, some users may wish to replace rules such as `http_archive` with their own versions that behave slightly differently or provide functionality that the regular versions do not.

When writing custom rules or functions, it is common to prefix their names with an underscore to make them private. Therefore, this change extends the Bazel manager to recognize rules beginning with an underscore, e.g. `_http_archive` as well as `http_archive`.

This change also fixes a small bug in `supportedRulesRegex` where grouping is not used around the `|` operator. This means that `^` and `$` are only included in the first and last alternatives respectively.

## Documentation

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/bazel/__fixtures__/WORKSPACE2
Comment (uhthomas) [inline lib/modules/manager/bazel/__fixtures__/WORKSPACE2:1]: Is there also another WORKSPACE which has the original `http_archive` rule? Just to make sure both it and the `_` prefixed version work.

Comment (zakcutner) [reply lib/modules/manager/bazel/__fixtures__/WORKSPACE2:1]: Yes! There is also `WORKSPACE1` and `WORKSPACE3`, which I didn't change

File: lib/modules/manager/bazel/rules/index.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ const supportedRules = [
   ...httpRules,
   ...mavenRules,
 ];
-export const supportedRulesRegex = regEx(`^${supportedRules.join('|')}$`);
+export const supportedRulesRegex = regEx(`^(?:${supportedRules.join('|')})$`);
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/index.ts:33]: why this change?

Comment (zakcutner) [reply lib/modules/manager/bazel/rules/index.ts:33]: Apologies, I forgot to mention this change in the description—I'll add something now. It's a small bug fix because `^a|b$` is interpreted as `(^a)|(b$)` rather than `^(a|b)$`. The use of `(?:)` rather than `()` just means that we use a non-matching group, which will likely be slightly more efficient as the regex engine will not need to record where the match happened.

**Review by viceice — APPROVED** [review]
